hera_cal/reflections.py

```diff
@@ -225,34 +157,63 @@ def model_reflections(self, dly_range, keys=None, data='clean', edgecut=0, taper
 
             # find reflection
             echo("...Modeling reflections in {}, assigning to {}".format(k, rkey), verbose=verbose)
-            (eps, amp, delays, phs, inds, sig, dfft,
-             dly_arr) = fit_reflection(data[k], dly_range, freqs, edgecut=edgecut, taper=taper,
-                                       zero_pad=zero_pad, full_freqs=self.freqs, fthin=fthin, **win_kwargs)
-            self.epsilon[rkey] = eps
-            self.amps[rkey] = amp
-            self.phs[rkey] = phs
-            self.delays[rkey] = delays
-            self.peak_ratio[rkey] = sig
-            self.dfft[k] = dfft
-
-    def write_reflections(self, output_calfits, input_calfits=None, overwrite=False, add_to_history=''):
+
+            # fit for reflection delays and amplitude
+            amp, dly, inds, sig = fit_reflection_delay(self.dfft[k], dly_range, self.delays, return_peak=False)
+
+            # make significance cut
+            if np.max(sig) < ref_sig_cut:
+                continue
+
+            # solve for phase (slowest step)
+            phs = fit_reflection_phase(self.dfft[k], dly_range, self.delays, amp, dly, fthin=fthin, Nphs=Nphs, ifft=True, ifftshift=True)
+
+            # anchor phase to 0 Hz
+            phs = (phs - 2 * np.pi * dly / 1e9 * (self.freqs[0] - zeropad * self.dnu)) % (2 * np.pi)
+
+            # form epsilon term
+            eps = construct_reflection(self.freqs, amp, dly / 1e9, phs, real=False)
+
+            self.ref_eps[rkey] = eps
+            self.ref_amp[rkey] = amp
+            self.ref_phs[rkey] = phs
+            self.ref_dly[rkey] = dly
+            self.ref_significance[rkey] = sig
+            self.ref_flags[rkey] = np.min(flags[k], axis=1, keepdims=True)
+
+        # form gains
+        self.ref_gains = form_gains(self.ref_eps)
+        for a in self.ants:
+            for p in self.pols:
+                k = (a, uvutils.parse_jpolstr(p)) 
+                if k not in self.ref_gains:
+                    self.ref_gains[k] = np.ones((self.Ntimes, self.Nfreqs), dtype=np.complex)
+
+    def write_auto_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+                               write_npz=False, add_to_history=''):
         """
+        Write auto reflection gain terms from self.ref_gains.
+
         Given a filepath to antenna gain calfits file, load the
```

Comment:
This should clarify that if input_calfits is None, the output is only reflections (especially this this the mode we plan on using this for in IDR2.2)

Reply:
Also, clarify where the flags/quals come from in both the case where you have an input_calfits and the case where you don't.

Reply:
I'll update the doc string to make it more clear in general